src/main/java/in/org/projecteka/hiu/dataflow/DataFlowController.java

```diff
@@ -20,8 +20,8 @@
     @PostMapping("/data/notification")
     public Mono<Void> dataNotification(@RequestBody DataNotificationRequest dataNotificationRequest) {
         return ReactiveSecurityContextHolder.getContext()
-                .map(securityContext -> (Caller) securityContext.getAuthentication().getPrincipal())
-                .map(Caller::getUsername)
+                .map(securityContext -> (GatewayCaller) securityContext.getAuthentication().getPrincipal())
+                .map(GatewayCaller::getUsername)
                 .flatMap(requesterId -> dataFlowService.handleNotification(dataNotificationRequest, requesterId));
```

Comment:
If /data/notification API is not called by gateway.. u can use caller only instead of Gateway caller

Reply:
renamed it to gateway caller